spec/plugin/out_rename_key_spec.rb

```diff
@@ -67,10 +67,30 @@ def create_driver conf=CONFIG, tag='test'
   context "private methods" do
     describe "#parse_rename_rule" do
       let(:rename_rule_example) { '^\$(.+) ${md[1]}' }
-      let(:result) { Fluent::RenameKeyOutput.new.parse_rename_rule rename_rule_example }
+      let(:rename_rule_result) { Fluent::RenameKeyOutput.new.parse_rename_rule rename_rule_example }
 
       it "captures 2 items, the key_regexp and new_name" do
-        expect(result).to have(2).items
+        expect(rename_rule_result.size).to eq 2
+      end
+    end
+
+    describe "#parse_replace_rule" do
+      let(:replace_rule_example1) { '- _' } # Replace hyphens with underscores
+      let(:replace_rule_result1) { Fluent::RenameKeyOutput.new.parse_replace_rule replace_rule_example1 }
+
+      let(:replace_rule_example2) { '[()-\s]' } # Remove all parethesis hyphens and spaces
```

Comment:
@perrin4869  Hi, I just noticed this regexp `[()-\s]` is causing some problem while refactoring the code. I think the hyphen makes ruby process it as from `)` to `\`. In the original tests, the string is not converted to regexp object so they didn't fail. There's no failing bug, but if you want to match all parentheses, hyphens, and spaces in your real code, try `[()\s-]` instead.